.github/workflows-helpers/run-e2e-test.sh

```diff
@@ -1,19 +1,28 @@
 # TODO_TECHDEBT(@okdas): also check readiness of appgate and relayminer to avoid false negatives due to race-conditions
 
-# Check if the pod with the matching image SHA and purpose is ready
-echo "Checking for ready validator pod with image SHA ${IMAGE_TAG}..."
+# Check if the pod with the matching image SHA and purpose is ready or needs recreation
+echo "Checking for ready validator pod with image SHA ${IMAGE_TAG} or pods needing recreation..."
 while :; do
     # Get all pods with the matching purpose
     PODS_JSON=$(kubectl get pods -n ${NAMESPACE} -l pokt.network/purpose=validator -o json)
 
     # Check if any pods are running and have the correct image SHA
     READY_POD=$(echo $PODS_JSON | jq -r ".items[] | select(.status.phase == \"Running\") | select(.spec.containers[].image | contains(\"${IMAGE_TAG}\")) | .metadata.name")
+    
+    # Check for non-running pods with incorrect image SHA to delete
+    INCORRECT_POD=$(echo $PODS_JSON | jq -r ".items[] | select(.status.phase != \"Running\") | select(.spec.containers[].image | contains(\"${IMAGE_TAG}\") | not) | .metadata.name")
 
     if [[ -n "${READY_POD}" ]]; then
         echo "Ready pod found: ${READY_POD}"
         break
+    elif [[ -n "${INCORRECT_POD}" ]]; then
+        echo "Non-ready pod with incorrect image found: ${INCORRECT_POD}. Deleting..."
+        kubectl delete pod -n ${NAMESPACE} ${INCORRECT_POD}
+        echo "Pod deleted. StatefulSet will recreate the pod."
+        # You may want to wait a bit here before the next iteration
```

Comment:
Can you update the comment explainin WHY we want to wait?

Reply:
@Olshansk oh, I updated comment. Thanks!